pkg/auth/auth.go

```diff
@@ -1,66 +1,137 @@
 package auth
```

Comment:
`auth` usually stands for `authentication`. Maybe we should call the package `authz` instead?